### ISSUE TITLE
HTTP 500 when sending request to mailgun

### ISSUE THREAD

**Issue by benoit74** [openzim/zimit-frontend#68]
Logs:

```
2024-07-30 06:59:26.049	DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.mailgun.net:443
2024-07-30 06:59:29.320	DEBUG:urllib3.connectionpool:https://api.mailgun.net:443 "POST /v3/mg.zimit.kiwix.org/messages HTTP/1.1" 500 36
2024-07-30 06:59:29.322	ERROR:utils:Failed to send mailgun notif: 500 Server Error: Internal Server Error for url: https://api.mailgun.net/v3/mg.zimit.kiwix.org/messages
2024-07-30 06:59:29.324	ERROR:utils:500 Server Error: Internal Server Error for url: https://api.mailgun.net/v3/mg.zimit.kiwix.org/messages
2024-07-30 06:59:29.324	Traceback (most recent call last):
2024-07-30 06:59:29.324	  File "/app/./utils.py", line 82, in send_email_via_mailgun
2024-07-30 06:59:29.324	    resp.raise_for_status()
2024-07-30 06:59:29.324	  File "/usr/local/lib/python3.8/site-packages/requests/models.py", line 943, in raise_for_status
2024-07-30 06:59:29.324	    raise HTTPError(http_error_msg, response=self)
2024-07-30 06:59:29.324	requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://api.mailgun.net/v3/mg.zimit.kiwix.org/messages
```

This is the first known occurrence AFAIK, maybe an issue on their side. Should we retry these calls?

**Comment by rgaudin** [comment]
I guess we should

**Comment by hamoudak** [comment]
I had the same issue for a website for two days or three when requesting on zimit.kiwix.org. when getting the zim file; not all pages working. so I have tried with the website in a later time, I got one working , but still  for now the same problem.

this is when browsing it ; some pages working till I get this issue.
<img width="956" alt="Untitled" src="https://github.com/user-attachments/assets/c37faf28-9b2d-4ffc-b7c1-e2a370a2f622">

**Comment by kelson42** [comment]
@benoit74 Can we secure this is correctly caught by the monitoring (if not already the case) so we are informed immediately next time?

**Comment by rgaudin** [comment]
@hamoudak you're saying that you had the same issue but this is not a user-visible issue. It only appears in logs.
Your screenshot is indeed an HTTP 500 one but for another website so clearly you're off here.

If you have an issue with zimit or zimit.kiwix.org, please open a new issue at [zimit repo](https://github.com/openzim/zimit/issues/new) clearly describing what you attempted and what you got.

@kelson42 please read ticket carefully ; this is not what you think it is.

**Comment by kelson42** [comment]
@rgaudin Read it twice abd still don't really understand why this issue is open 2 months after it occured... therefore my question: are such errors monitored?

**Comment by benoit74** [comment]
And it is still opened because I have no idea of what to do when mailgun API answers with a 500 error. And since it did not reproduced again, I consider it is a very rare edge case not worth to work on.

**Comment by kelson42** [comment]
> And it is still opened because I have no idea of what to do when mailgun API answers with a 500 error.

Opening an issue immediatly does not imply we know how to handle it. For the rest opening a support case seems the good approach if we are cluless.

**Comment by benoit74** [comment]
I won't open a support case for something which happened only once. I know in advance how it will finish. Feel free to contact Mailgun if you wish.

**Comment by kelson42** [comment]
We can close then and reopen if necessary IMHO.
